backend/src/graphql/resolver/util/findUserByIdentifier.ts

```diff
@@ -0,0 +1,35 @@
+import { User as DbUser } from '@entity/User'
+import { UserContact as DbUserContact } from '@entity/UserContact'
+import LogError from '@/server/LogError'
+import { validate, version } from 'uuid'
+
+export const findUserByIdentifier = async (identifier: string): Promise<DbUser | null> => {
+  let user: DbUser | undefined
```

Comment:
I am a big fan of success case last and error cases before, but here it clearly has disadvantages.

I suggest to treat this like a switch case and return immediately after you have found an user. This will make it easy to reduce the result to `Promise<DbUser>` and simplify the resolver function like suggested above.